hazelcast-jet-core/src/main/java/com/hazelcast/jet/Observable.java

```diff
@@ -0,0 +1,222 @@
+
+/**
+ * Represents a flowing sequence of events produced an
+ * {@link Sinks#observable(String) observable sinks}. To observe the events,
+ * call {@link #addObserver jet.getObservable(name).addObserver(myObserver)}.
+ * <p>
+ * The {@code Observable} is backed by a {@link Ringbuffer}, which, once
+ * created, has a fixed capacity for storing messages. It supports reading
+ * by multiple {@code Observer}s, which will all observe the same sequence of
+ * messages. A new {@code Observers} will start reading automatically from the
+ * oldest sequence available. Once the capacity is full, the oldest messages
+ * will be overwritten as new ones arrive.
+ * <p>
+ * The {@code Ringbuffer}'s capacity defaults to 10,000, but can be changed
+ * (via the {@link #setCapacity(int) setCapacity()} method), as long as the
+ * {@code Ringbuffer} hasn't been created yet (see the "Lifecycle" section below).
+ * <p>
+ * In addition to data events, the {@code Observer} can also observe
+ * completion and failure events. Completion means that no further values
+ * will appear in the sequence. Failure means that something went wrong
+ * during the production of the sequence's values and the event attempts
+ * to provide useful information about the cause of the problem.
+ * <p>
+ * <strong>Lifecycle</strong>
+ * <p>
+ * When talking about the lifecycle of an {@code Observable} (which is
+ * basically just a client side object and has a lifecycle just like any
+ * other POJO) it's better to actually consider the lifecycle of the
+ * underlying {@code Ringbuffer}, since that is the significant
+ * distributed entity.
+ * <p>
+ * The lifecycle of the {@code Ringbuffer} is decoupled from the lifecycle
+ * of the job. The {@code Ringbuffer} is created either when the user
+ * gets a reference to its equivalent {@code Observable} (through
+ * {@link JetInstance#getObservable(String) JetInstance.getObservable()})
+ * and registers the first {@link Observer} on it (through
+ * {@link Observable#addObserver(Observer) Observable.addObserver()})
+ * or when the job containing the sink for it starts executing.
+ * <p>
+ * The {@code Ringbuffer} must be explicitly destroyed when it's no longer
+ * in use, or data will be retained in the cluster. This is done via the
+ * {@link #destroy() Observable.destroy()} method. Note: even if the
+ * {@code Observable} POJO gets lost and its underlying {@code Ringbuffer}
+ * floats around in the cluster, it's still possible to manually destroy
+ * it later by creating another {@code Observable} instance with the same
+ * name and calling {@code destroy()} on that.
+ * <p>
+ * <strong>Important:</strong> The same {@code Observable} must
+ * <strong>not</strong> be used again in a new job since this will cause
+ * completion events interleaving and causing data loss or other unexpected
+ * behaviour. Same goes for using one observable name in multiple
+ * {@link Sinks#observable(String) observable sinks} in the same job.
```

Comment:
I don't think this is accurate, it should be fine to use the same observables in the same job for multiple sinks. There is only one completion event for the whole job.